app/scripts/services/formatting.js

```diff
@@ -135,24 +135,21 @@ angular.module('dockstore.ui')
       };
 
       this.getImageReposProvider = function(path) {
-        if (path.indexOf('quay.io') !== -1) {
-          return 'QUAY_IO';
-        } else if (path.indexOf('hub.docker.com') !== -1) {
-          return 'DOCKER_HUB';
-        } else {
-          return null;
+        for (var i = 0; i < dockerRegistryMap.length; i++) {
+          if (path.indexOf(dockerRegistryMap[i].dockerCommand) !== -1) {
+            return dockerRegistryMap[i].enum;
+          }
         }
+        return null;
       };
 
       this.getImageReposProviderName = function(imageReposProvider) {
-        switch (imageReposProvider) {
-          case 'QUAY_IO':
-            return 'Quay.io';
-          case 'DOCKER_HUB':
-            return 'Docker Hub';
-          default:
-            return 'Unknown';
+        for (var i = 0; i < dockerRegistryMap.length; i++) {
+          if (imageReposProvider === dockerRegistryMap[i].enum) {
+            return dockerRegistryMap[i].friendlyName;
+          }
         }
+        return 'Unknown';
       };
 
       this.getImageReposWebUrl = function(path, imageReposProvider) {
```

Comment:
I can work on a new branch once this one is done and go through all the javascript files and add documentation.